src/lib/provider-wrapper-utils.ts

```diff
@@ -0,0 +1,42 @@
+import * as abi from 'web3-eth-abi';
+import { JsonRpc } from '../types/JsonRpc';
+import { Method } from '../types/Method';
+import { METHODS } from './constants';
+
+let idCount = 0;
+// @ts-ignore
+const web3abiDecoder = abi.default;
+
+export function decodeResult(method: Method, result) {
+  const rpcResult = result.result;
+  return rpcResult === '0x'
+    ? null
+    : web3abiDecoder.decodeParameter(METHODS[method].returnEncoding, rpcResult);
+}
```

Comment:
This lib is typed so I don't understand we are not getting the types.

I tried locally and this seems to preserve the types:

![image](https://user-images.githubusercontent.com/477945/129565351-342cf4a1-e598-4265-9401-699635954455.png)

Not sure tho if it has the same behaviour as idk what `abi.default` returns

Reply:
I have already tried this. I'm afraid this does not work, it does compile, but It then fails at runtime with:

```bash
TypeError: web3_eth_abi_1.AbiCoder is not a constructor
```

Maybe we can look into it together?

The package actually exports an instance, so we are not supposed to call it with `new`. When trying to use it as an instance, compilation fails.